requests/models.py

```diff
@@ -932,6 +944,8 @@ def raise_for_status(self):
             http_error_msg = u'%s Server Error: %s for url: %s' % (self.status_code, reason, self.url)
 
         if http_error_msg:
+            if isinstance(body_text, basestring):
+                http_error_msg += u' Response Body: %s' % body_text
```

Comment:
The body text may be *enormous*: megabytes in size. We absolutely do not want to build a string that long. If we're adding the body text to the exception message, we should add only a segment of it.